Kudu.Core/SiteExtensions/SiteExtensionManager.cs

```diff
@@ -449,33 +451,46 @@ private SiteExtensionInfo GetPreInstalledExtension(string id)
                 string packageLocalFilePath = GetNuGetPackageFile(package.Identity.Id, package.Identity.Version.ToString());
                 using (tracer.Step("Download site extension: {0}", package.Identity))
                 {
-                    await remoteRepo.DownloadPackageToFolder(package.Identity, installationDirectory, pathToLocalCopyOfNudpk: packageLocalFilePath);
-                }
+                    string extractPath = installationDirectory;
+                    if (string.Equals(SiteExtensionInfo.SiteExtensionType.WebRoot, type, StringComparison.OrdinalIgnoreCase))
+                    {
+                        extractPath = _environment.WebRootPath;
+                        FileSystemHelpers.EnsureDirectory(extractPath);
+                    }
 
-                // If there is no xdt file, generate default.
-                using (tracer.Step("Check if applicationhost.xdt file existed."))
-                {
-                    GenerateApplicationHostXdt(installationDirectory, '/' + package.Identity.Id, isPreInstalled: false, tracer: tracer);
+                    await remoteRepo.DownloadPackageToFolder(package.Identity, extractPath, pathToLocalCopyOfNudpk: packageLocalFilePath);
+                    tracer.Trace("Generate xdt file to map application to run from wwwroot when request for /{0}", package.Identity.Id);
+                    GenerateRemapToWebRootApplicationHostXdt(installationDirectory, string.Format(CultureInfo.InvariantCulture, "/{0}", package.Identity.Id));
```

Comment:
It looks like this always generate it, even in the non-WebRoot case. Doesn't that break things for the general case of a site extension that has an xdt?